content/en-US/api/URLPattern.md

```diff
@@ -0,0 +1,515 @@
+// the brackets.
+// 2. Answer all questions in comments.
+// 3. Remove all comments.
+
+**When this feature ships, the content below will live on MDN under
+[developer.mozilla.org/en-US/docs/Web/API](https://developer.mozilla.org/en-US/docs/Web/API).**
+
+// Example interface page at: https://developer.mozilla.org/en-US/docs/Web/API/ContentIndex
+
+## Description
+
+// Complete the sentence below with a basic description of the API.
+// You may add more sentences if needed.
+
+The `URLPattern` interface provides a web platform primitive for matching
+URLs.  The pattern syntax is adopted from the popular [path-to-regexp][]
+javascript library and is more ergonomic than using regular expressions.
+
+## Constructor
+
+Creates a new `URLPattern` object.
+
+## Properties
+
+**`URLPattern.protocol`**
+
+Returns the URL protocol pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.username`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.password`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.hostname`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.port`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.pathname`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.search`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.hash`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+## Events
+
+None
+
+## Methods
+
+**`URLPattern.test()`**
+
+Matches the input against the pattern returning true on success and false
+otherwise.
+
+**`URLPattern.exec()`**
+
+Matches the input against the pattern returning a URLPatternResult object on
+success and null otherwise.
+
+## Examples
+
+### Example: Construct a URLPattern to filter on a specific URL component.
+
+The following example shows how a URLPattern can be used to filter on a
+specific URL component.  When the URLPattern constructor is called with a
+structured object of component patterns any missing components default to
+the `*` wildcard value.
+
+```js
+// Construct a URLPattern that matches a specific domain and its subdomains.
+// All other URL components default to the wildcard `*` pattern.
+const pattern = new URLPattern({
+  hostname: {*.}?example.com
```

Comment:
Does the value of `hostname` need to be in quotes? When I run this in a console, the period causes 'SyntaxError: Unexpected token'.

Reply:
No.  But the pattern itself needs to be in quotes.  Look like I forgot that here.

Reply:
That's what I meant: the pattern.

Reply:
Fixed in latest push